File: src-tauri/src/omp_auth.rs
//! omp CLI 凭据读取 —— agent.db(auth_credentials 表)只读访问。
//!
//! omp 凭据存 sqlite(~/.omp/agent/agent.db),JS 无法解析,由 Rust 代读。
//! 这是 CLI 私有存储知识,独立成模块:quota.rs 保持"通用 HTTP 代理 +
//! 只读环境变量"的纯粹职责(见 docs/architecture/01-overview.md §7)。

/// 读取 omp CLI 某供应商的最新凭据 data JSON(auth_credentials 表)。
/// omp 凭据存 sqlite(~/.omp/agent/agent.db),JS 无法解析,由 Rust 只读取出。
/// 库不存在/无记录返回 Ok(None),不抛错(前端据此显示空态)。
#[tauri::command]
pub fn omp_auth_credential(provider: String) -> Result<Option<String>, String> {
    let db_path = dirs::home_dir()
        .unwrap_or_else(std::env::temp_dir)
        .join(".omp/agent/agent.db");
    if !db_path.exists() {
        return Ok(None);
    }
    let conn =
        rusqlite::Connection::open_with_flags(&db_path, rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY)
            .map_err(|e| format!("open omp agent.db: {e}"))?;
    let mut stmt = conn
        .prepare(
            "SELECT data FROM auth_credentials \
             WHERE provider = ?1 AND disabled_cause IS NULL \
             ORDER BY updated_at DESC LIMIT 1",
        )
        .map_err(|e| format!("prepare omp credential query: {e}"))?;
    let mut rows = stmt
        .query(rusqlite::params![provider])
        .map_err(|e| format!("query omp credential: {e}"))?;
    match rows
        .next()
        .map_err(|e| format!("read omp credential row: {e}"))?
    {
        Some(row) => {
            let data: String = row
                .get(0)
                .map_err(|e| format!("read credential data column: {e}"))?;
            Ok(Some(data))
        }
        None => Ok(None),
    }
}
/// 列出 omp CLI 已登录的全部供应商 id(auth_credentials 表,未禁用)。
/// 用于欢迎页"已登录供应商"盘点;库不存在/查询失败返回空表,不抛错。
#[tauri::command]
pub fn omp_auth_providers() -> Vec<String> {
    let db_path = dirs::home_dir()
        .unwrap_or_else(std::env::temp_dir)
        .join(".omp/agent/agent.db");
    if !db_path.exists() {
        return Vec::new();
    }
    let Ok(conn) =
        rusqlite::Connection::open_with_flags(&db_path, rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY)
    else {
        return Vec::new();
    };
    let Ok(mut stmt) = conn.prepare(
        "SELECT DISTINCT provider FROM auth_credentials \
         WHERE disabled_cause IS NULL ORDER BY provider",
    ) else {
        return Vec::new();
    };
    let rows = stmt.query_map([], |row| row.get::<_, String>(0));
    match rows {
        Ok(mapped) => mapped.filter_map(Result::ok).collect(),
        Err(_) => Vec::new(),
    }
}

File: src/plugins/cli-shared/quota/ompAuth.ts
/**
 * omp agent.db 凭据读取 —— omp CLI 私有库知识(~/.omp/agent/agent.db,
 * auth_credentials 表)集中地。
 *
 * 消费者:cli-omp(quota provider)+ welcome(引擎卡已登录供应商盘点),
 * 满足 cli-shared 准入(≥2 插件消费同一 CLI 磁盘格式)。
 * 经 kernel 的通用只读 sqlite 原语(ipc.sqliteQuery)代读,内核不理解本格式。
 */

import { ipc } from "@kernel/ipc";

/** omp 凭据库相对家目录路径。 */
const OMP_AGENT_DB = ".omp/agent/agent.db";

async function ompDbPath(): Promise<string> {
  const home = await ipc.configHomeDir();
  return `${home.replace(/\/$/, "")}/${OMP_AGENT_DB}`;
}

function cellText(value: unknown): string | null {
  return typeof value === "string" ? value : null;
}

/** 列出已登录且未禁用的供应商 id(auth_credentials 表,按 id 升序)。
 *  库不存在/查询失败 = 空表,不抛错。 */
export async function listOmpAuthProviders(): Promise<string[]> {
  const rows = await ipc
    .sqliteQuery(
      await ompDbPath(),
      "SELECT DISTINCT provider FROM auth_credentials \
       WHERE disabled_cause IS NULL ORDER BY provider",
      [],
    )
    .catch(() => [] as unknown[][]);
  return rows
    .map((row) => cellText(row[0]))
    .filter((v): v is string => v !== null);
}

/** 读某供应商最新一条凭据的 data JSON(按 updated_at 取新,仅未禁用)。
 *  无记录 = null,不抛错。 */
export async function readOmpAuthCredential(
  provider: string,
): Promise<string | null> {
  const rows = await ipc
    .sqliteQuery(
      await ompDbPath(),
      "SELECT data FROM auth_credentials \
       WHERE provider = ?1 AND disabled_cause IS NULL \
       ORDER BY updated_at DESC LIMIT 1",
      [provider],
    )
    .catch(() => [] as unknown[][]);
  return rows[0] ? cellText(rows[0][0]) : null;
}
